pkg/autopilot/common/telemetry.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 k0s authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"github.com/k0sproject/k0s/pkg/telemetry"
+	"github.com/segmentio/analytics-go"
+)
+
+const eventName = "autopilotTransitionEvent"
+
+var reporter = telemetry.NewDefaultSegmentClient()
+
+type Event struct {
+	ClusterID string
+	OldStatus string
+	NewStatus string
+}
+
+func ReportEvent(evt *Event) error {
+	if reporter == nil {
+		return nil
+	}
+	return reporter.Enqueue(analytics.Track{
+		Event:      eventName,
+		Properties: evt.asProperties(),
+	})
+}
+
+func (e *Event) asProperties() analytics.Properties {
+	return analytics.Properties{
+		"oldStatus": e.OldStatus,
+		"newStatus": e.NewStatus,
```

Comment:
cluster id field might be added here as well

Reply:
Great catch, thank you. Fixed